src/redux/stores/collection/collectionSlice.ts

```diff
@@ -21,27 +23,33 @@ const collectionSlice = createSlice({
   initialState,
   reducers: {},
   extraReducers: builder => {
-    builder.addCase(fetchNFTMetadata.pending, state => ({
+    builder.addCase(fetchCollectionTokens.pending, state => ({
       ...state,
       isLoading: true,
     }));
 
-    builder.addCase(fetchNFTMetadata.fulfilled, (state, action) => {
+    builder.addCase(fetchCollectionTokens.fulfilled, (state, action) => {
+      // When fetchNFTMetadata returns undefined for a token we assume that's the end of the token list
+      // TODO: We might want to revise this when we add support for irregular token id's. https://github.com/airswap/airswap-marketplace/issues/49
+      const allTokensAreLoaded = action.payload.includes(undefined);
+      const filteredTokens = action.payload.filter(token => token !== undefined) as CollectionToken[];
+
       const newTokensData = getUniqueArrayChildren([
         ...state.tokensData,
-        ...action.payload,
+        ...filteredTokens,
       ].sort((a, b) => a.id - b.id), 'id') as CollectionToken[];
       const lastToken = newTokensData[newTokensData.length - 1];
 
       return {
         ...state,
+        allTokensAreLoaded,
         isLoading: false,
         tokensData: newTokensData,
         lastTokenIndex: lastToken ? lastToken.id : 1,
       };
     });
 
-    // TODO: Add error handling
+    // TODO: Add error handling https://github.com/airswap/airswap-marketplace/issues/48
     // builder.addCase(fetchNFTMetadata.rejected, (state, action) => {
```

Comment:
I'd sooner see a simple console log left here, I have actually suffered from the lack of this log and added it in my own branch

```suggestion
     builder.addCase(fetchNFTMetadata.rejected, (state, action) => {
      console.log('fetchNFTMetadata.rejected', action);
    });
```

Reply:
Will do!